### PR TITLE
Bugfix for survey species list and added session and location functions

### PR DESCRIPTION
**Pull request by MrDrummer**

### CODE REVIEW

File: central_web_server/ui_files/survey.js [hunk 1]
```diff
@@ -30,8 +30,14 @@ function logHttpStateChange(http) {
 }
 
 function submit_onClick() {
+    let foundSpeciesObjects = [];
+    for (let species of foundSpecies) {
+      foundSpeciesObjects.push({
+        "species_id": species
+      })
```
Comment (ghost) [inline central_web_server/ui_files/survey.js]: semicolons for consistency

File: central_web_server/handle_database.js [hunk 2]
```diff
@@ -110,13 +113,38 @@ async function addSurveyResults(surveyData) {
...
     .setFieldsRows(surveyResults)
     .toString()
   await connection.query(surveyResultsQuery)
 }
+addSurveyResults ({
+	"species_list_id" : 4,
+	"tourist_id" : ["kh39b","jhu89"],
+	"session_id" : "0g55l",
+	"found_species" :[
+    {
```
Comment (ghost) [inline central_web_server/handle_database.js]: oneline these for readability (only if you think that looks better)
ie:
```
{ "species_id": 1, },
{ "species_id": 3, },
{ "species_id": 6, },
{ "species_id": 7, }
```

File: central_web_server/ui_files/survey.js [hunk 3]
```diff
@@ -30,8 +30,14 @@ function logHttpStateChange(http) {
 }
 
 function submit_onClick() {
+    let foundSpeciesObjects = [];
```
Comment (ghost) [inline central_web_server/ui_files/survey.js]: objects?
maybe `submittedSpecies` is a more understandable name?

**Review by ghost — CHANGES_REQUESTED** [review]
just some readability stuff

File: central_web_server/handle_database.js [hunk 4]
```diff
@@ -114,9 +113,45 @@ async function addSurveyResults(surveyData) {
...
+  let getLocationQuery = squel
+    .select()
+    .from("location")
+    .toString();
+  let getLocationResult = await connection.query(getLocationQuery);
+  console.log(getLocationResult);
+  connection.end();
+}
+addLocation([{"name": "Mars"}]);
+getLocation();
```
Comment (byran) [inline central_web_server/handle_database.js]: Remove the two lines above as they will create things every time the script is imported

File: database/setup-database.sql [hunk 5]
```diff
@@ -11,11 +11,12 @@ CREATE TABLE survey_results (
 CREATE TABLE session (
   id int AUTO_INCREMENT PRIMARY KEY,
   date datetime NOT NULL,
-  location varchar(20) NOT NULL,
+  location_id varchar(5) NOT NULL,
```
Comment (byran) [inline database/setup-database.sql]: I think this wants to be an int, not a varchar as it's an index to the location table

File: database/setup-database.sql [hunk 6]
```diff
@@ -11,11 +11,12 @@ CREATE TABLE survey_results (
...
   time_low_tide time NOT NULL,
   height_low_tide tinyint NOT NULL,
+  temperature tinyint NOT NULL,
```
Comment (byran) [inline database/setup-database.sql]: I think this should be a float or double

**Review by byran — CHANGES_REQUESTED** [review]

**Pull request merged**